pkg/operator/configobserver/config_observer_controller.go

```diff
@@ -135,6 +153,50 @@ func (c ConfigObserver) sync(ctx context.Context, syncCtx factory.SyncContext) e
 	return configError
 }
 
+func (c ConfigObserver) updateObservedConfig(syncCtx factory.SyncContext, existingConfig map[string]interface{}, mergedObservedConfig map[string]interface{}) error {
```

Comment:
I think something like this would help me have parsed this.

```go
func() {
  if c.nestedConfigPath == 0 {
    if !deepEqual(existingConfig, mergedObservedConfig){
      return c.updateConfig(mergedObservedConfig)
    }
    return nil
  }

  // do the other stuff
}

func (c ConfigObserver) updateConfig(config map) error{
  if _, _, err := v1helpers.UpdateSpec(c.operatorClient, v1helpers.UpdateObservedConfigFn(updatedConfig)); err != nil {
    // At this point we failed to write the updated config. If we are permanently broken, do not pile the errors from observers
    // but instead reset the errors and only report single error condition.
    syncCtx.Recorder().Warningf("ObservedConfigWriteError", "Failed to write observed config: %v", err)
    return fmt.Errorf("error writing updated observed config: %v", err)
  }
  return nil
}

```

Reply:
done